ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java

```diff
@@ -743,7 +751,11 @@ public void communicatorPausedOnError() {
     }
 
     public void checkStreamFinished() {
-        if (this.isStreaming() && !this.comm.areActiveCommands() && this.comm.numActiveCommands() == 0) {
+        if (this.isStreaming() &&
+                !this.comm.areActiveCommands() &&
+                this.comm.numActiveCommands() == 0 &&
+                rowsRemaining() <= 0 &&
+                (getControlState() == COMM_IDLE || getControlState() == COMM_SENDING_PAUSED)) {
```

Comment:
When does `COMM_SENDING_PAUSED` need to be checked?

Reply:
Oh yeah, I think that was for Pre-Feed-Hold grbl where pause just stopped sending commands and let the stream proceed to an idle state.

Reply:
So it's ok to merge?

Reply:
yes